aip/general/0236.md

```diff
@@ -0,0 +1,315 @@
+    start or stop downstream evaluation using custom methods described below.
+- The first time the startPreview custom method is used, the service **must**
+  create `preview_metadata` and do the following: 
+  - It **must** set the `state` to ACTIVE
+  - It **must** set a `log_prefix` string, which is a predefined constant set to
+    the same value for all preview mode logs for the given resource type.
+  - It **must** populate the `start_time` with the current time. 
+- When the stopPreview custom method is used, the service **must** do the
+  following:
+  - It **must** set the `state` to INACTIVE
+  - It **must** populate the `stop_time` with the current time. 
+- `start_time` **must** be updated every time the status is changed to ACTIVE.
+
+### APIs
+
+Experiments **must** follow resource-oriented design as stated in
+([AIP-121][aip-121]). Additionally, the following guidance **must** be followed
+which is specific to experiments:
+
+#### create
+
+- Creating a new experiment for preview mode **must** support the following
+use cases:
+  - Preview a new Policy.
+  - Preview an update to an already live Policy.
+  - Preview a deletion of a current Policy.
+- For the update or delete use cases, the `policy` field in the `experiment`
+  **must** have the full payload of the live policy copied into it, including
+  the name. 
+  - The fields under test **must** be updated to the new intended state to 
+    preview an update.
+  - The `rules` **must** be set to empty to preview a delete.
+- Previewing the creation of a brand new Policy **must** be supported. 
+  - If the service does not support a nested collection without a live policy, 
+    the user **must** create a top-level resource which has the correct rules to
+    represent a no-op. Then, experiments can be added under that resource.
+- If multiple previews are supported, calling the create more than once on a
+  resource **must** lead to the creation of multiple experiments which are
+  represented as new subresources until the set quota is reached.
+
+#### update
+
+- The name inside “Policy” **must not** be changed.
+- The service **must** set the `state` to INACTIVE if the `state` was ACTIVE at
+  the time of an update. This is so the user can easily distinguish between 
+  previews of different fields-under-test.
+
+#### list
+
+- Filtering on `PreviewMetadata.state` **must** indicate which experiments are
+  actively being previewed or not.
+
+### Live policy API changes
+
+The following guidance **must** be followed for live resources which use
+experiments as a nested collection for preview.
+
+#### delete
+
+- A delete of the top-level resource **must** delete all experiments.
+- To maintain the experiments, the top-level Policy **must** be changed to a
+  no-op Policy instead of using delete.
+
+### Custom method APIs
+
+#### startPreview
+
+```proto
+rpc StartPreviewPolicyExperiment(StartPreviewPolicyExperimentRequest) returns (google.longrunning.Operation) {
```

Comment:
This method name does not comply with AIP-136. It should contain the resource name. The resources described herein are, `Policy` and `PolicyExperiment`. Can we try to come up with a custom method name pattern that is compliant with AIP-136?